developer/guides/components/demo-component.mdx

```diff
@@ -0,0 +1,559 @@
+
+### Accordion items
+
+We can access the Makeswift controls and have them return input from the user in Makeswift when we register the component in the `Accordions.makeswift.ts` file. Each control adds properties to the component that we can edit in Makeswift.
+
+We want the user to be able to edit the title for an accordion item. For this, we can use the [`TextInput`](/developer/reference/controls/text-input) control. We'll import this into the `Accordions.makeswift.ts` file and add it to our `props` object.
+
+```tsx src/components/Accordions/Accordions.makeswift.tsx {1, 11-14}
+import { TextInput } from "@makeswift/runtime/controls";
+
+import { runtime } from "src/makeswift/runtime";
+
+import { Accordions } from "./";
+
+runtime.registerComponent(Accordions, {
+  type: "accordions",
+  label: "Accordions",
+  props: {
+    accordions: TextInput({
+      label: "Title",
+      defaultValue: "Accordion title",
+    }),
+  },
+});
+```
+
+We also want the user to be able to add multiple accordion items, not just one. To support this, we can use the [`List`](/developer/reference/controls/list) control and put the `title` prop inside of it.
+
+```tsx src/components/Accordions/Accordions.makeswift.tsx {1, 11-17}
+import { TextInput, List } from "@makeswift/runtime/controls";
+
+import { runtime } from "@/makeswift/runtime";
+
+import { Accordions } from "./";
+
+runtime.registerComponent(Accordions, {
+  type: "accordions",
+  label: "Accordions",
+  props: {
+    accordions: List({
+      label: "Accordions",
+      type: TextInput({ label: "Title", defaultValue: "Accordion Title" }),
+      getItemLabel(accordionItem) {
+        return accordionItem ?? "Accordion Title";
+      },
+    }),
+  },
+});
+```
+
+Lastly, since we're going to add a `children` property to each accordion item next, we'll want to use the [`Group`](/developer/reference/controls/group) control to define each accordion item as an object. For the `Group` control we'll need to define its own props which will include the `title` props. We can also set its `preferredLayout` prop to be `Inline`.
+
+```tsx src/components/Accordions/Accordions.makeswift.tsx {3, 13-22, 24}
+import { runtime } from "src/makeswift/runtime";
+
+import { TextInput, List, Group } from "@makeswift/runtime/controls";
+
+import { Accordions } from "./";
+
+runtime.registerComponent(Accordions, {
+  type: "accordions",
+  label: "Accordions",
+  props: {
+    accordions: List({
+      label: "Accordions",
+      type: Group({
+        label: "Accordion",
+        preferredLayout: Group.Layout.Inline,
+        props: {
+          title: TextInput({
+            label: "Title",
+            defaultValue: "Accordion title",
+          }),
+        },
+      }),
+      getItemLabel(accordionItem) {
+        return accordionItem?.title || "Accordion title";
+      },
+    }),
+  },
+});
+```
+
+Next, adjust the `index.tsx` file to accept the `accordions` prop.
```

Comment:
To reiterate, IMHO it’s much easier to understand the connection between component props and their registration if you update the component first, then update the registration to reflect the new props.